testproject/Assets/Tests/Runtime/SenderIdTests.cs

```diff
@@ -89,36 +90,34 @@ public IEnumerator WhenSendingMessageFromClientToServer_SenderIdIsCorrect()
             Assert.IsTrue(secondReceived);
         }
 
+
+        private List<ulong> m_ClientsDisconnected = new List<ulong>();
+        private ulong m_ClientToValidateDisconnected;
+
+        private bool ValidateClientId()
+        {
+            return m_ClientsDisconnected.Contains(m_ClientToValidateDisconnected);
+        }
+
+        private void OnClientDisconnected(ulong clientId)
+        {
+            m_ClientsDisconnected.Add(clientId);
+        }
+
         [UnityTest]
         public IEnumerator WhenClientDisconnectsFromServer_ClientIdIsCorrect()
         {
-            var firstClientId = FirstClient.LocalClientId;
-            bool received = false;
-            void firstCallback(ulong id)
-            {
-                Assert.AreEqual(firstClientId, id);
-                received = true;
-            }
-            m_ServerNetworkManager.OnClientDisconnectCallback += firstCallback;
+            m_ClientsDisconnected.Clear();
+            m_ServerNetworkManager.OnClientDisconnectCallback += OnClientDisconnected;
```

Comment:
Can you also add a check for this behavior in `OnConnectionEvent`? `InvokeOnClientDisconnectCallback` should invoke both of them.

Reply:
Also discovered an issue with StopOneClient while adding the test... so it was a 2 4 1.
Validated the ConnectionEvent.ClientDisconnected triggers under both scenarios and fixed the integration test method.
👍